ios/RCTMapboxGL/RCTMapboxGL.m

```diff
@@ -72,7 +72,21 @@ - (void)updateMap
         _map.scrollEnabled = _scrollEnabled;
         _map.zoomEnabled = _zoomEnabled;
         _map.showsUserLocation = _showsUserLocation;
-        _map.styleURL = _styleURL;
+        if ([_styleURL  isEqual: @"lightStyleURL"]) {
```

Comment:
If the separate factory object idea doesn't fit too well in this project's architecture, there could be a standalone function that maps default style name strings like `"emerald"` to the corresponding mapbox: URLs. The developer would call that function and pass the return value into the `styleURL` property.